qiime/core/type/signature.py

```diff
@@ -113,12 +114,20 @@ def decode_parameters(self, **kwargs):
         return params
 
     def check_types(self, **kwargs):
-        for key, (type_, _) in itertools.chain(self.inputs.items(),
-                                               self.parameters.items()):
+        for key, (type_, _) in self.inputs.items():
+            if kwargs[key] is None:
+                raise TypeError("Argument to input %r is None, but only "
```

Comment:
Why is this check necessary? `None` isn't a subtype of a semantic type, just like `42` isn't. I don't think we need to special-case `None` here because it would fail below on line 121.